### PR TITLE
put all intake catalogs here

### PR DESCRIPTION
**Pull request by rabernat**
This PR moves all of the known pangeo intake catalogs into this repo.

TODO:

- [x] Figure out of anything is missing
- [x] Possibly replace relative paths in `master.yaml` with URLs?
- [ ] Decide how to handle `consolidated` option. (No released versions of xarray / zarr support it--would be nice if it could be optional.)
- [x] Add CI to make sure all catalog entries open properly.

cc @martindurant

### CODE REVIEW

**Comment by martindurant** [comment]
This is fine with me.
You may want to include a master catalog too, to explicitly point at the others and include a little descriptive text about what they are and where they are used.

**Comment by rabernat** [comment]
Specific question for @martindurant: If I open a catalog from a URL, i.e.
```python
cat = intake.Catalog('http://server/master.yaml')
```

and that file contains entries like
```
  ocean:
    args:
      path: ocean.yaml
    description: 'Pangeo Oceanography Dataset Catalog'
    driver: intake.catalog.local.YAMLFileCatalog
    metadata: {}
```

Will it figure out how to expand the path appropriately to also pull from the remote server? Or does `path` need to be `'http://server/ocean.yaml'?

**Comment by martindurant** [comment]
You can either include the absolute path, or use "{{CATALOG_DIR}}/ocean.yaml"

**Comment by rabernat** [comment]
> You may want to include a master catalog too

There is one! See `master.yaml`.

**Comment by martindurant** [comment]
> There is one

(sorry)

File: tests/validate_datasets.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import intake
+import pytest
+
+def crawl_nested_catalog(cat_or_entry, parent_name=()):
```
Comment (martindurant) [inline tests/validate_datasets.py]: May be simpler to use [`cat.walk`](https://intake.readthedocs.io/en/latest/api_user.html#intake.catalog.local.YAMLFileCatalog.walk)

**Pull request merged**